consensus/consensus-types/src/common.rs

```diff
@@ -33,29 +41,77 @@ pub struct RejectedTransactionSummary {
     pub hash: HashValue,
 }
 
+#[derive(Debug)]
+pub enum DataStatus {
+    Cached(Vec<SignedTransaction>),
+    Requested(Vec<oneshot::Receiver<Result<Vec<SignedTransaction>, Error>>>),
+}
+
+// impl PartialEq for DataStatus {
+//     fn eq(&self, other: &Self) -> bool {
+//         match (self, other) {
+//             (DataStatus::Cached(ltxn), DataStatus::Cached(rtxn)) => ltxn == rtxn,
+//             (DataStatus::Requested(lchannels), DataStatus::Requested(rchannels)) => Arc::as_ptr(&lchannels) == Arc::as_ptr(&rchannels),
+//             (_, _) => false,
+//         }
+//     }
+// }
+//
+// impl Eq for DataStatus {}
+
+#[derive(Deserialize, Serialize, Clone, Debug)]
+pub struct ProofWithStatus {
+    pub proofs: Vec<ProofOfStore>,
+    #[serde(skip)]
+    pub status: Arc<Mutex<Option<DataStatus>>>,
```

Comment:
I need to think about this a bit, it feels a bit awkward to have this being changed underneath.

Also is it guaranteed that after deserialization/inconstructor, statuses is correctly initialized with a new Mutex pointing to None?

Reply:
This is a change I made as part of the review. Before PayloadManager had a DashMap to keep track of Blocks statuses. Now, the PayloadManager code is cleaner, but the tradeoff is that we change the payload underneath.
Togther with @bchocho @gelash and @zekun000 we spent some time thinking about it and making it work.